sys/net/gnrc/network_layer/ipv6/gnrc_ipv6.c

```diff
@@ -852,7 +852,6 @@ static void _receive(gnrc_pktsnip_t *pkt)
         ipv6 = gnrc_pktbuf_mark(pkt, sizeof(ipv6_hdr_t), GNRC_NETTYPE_IPV6);
 
         first_ext = pkt;
-        pkt->type = GNRC_NETTYPE_UNDEF; /* snip is no longer IPv6 */
```

Comment:
The 6LoWPAN module creates packet like this:

![set_undef_after_mark_1](https://cloud.githubusercontent.com/assets/3772478/16573235/787f36dc-42ac-11e6-9baa-36ef1aa869fd.png)

then the IPv6 module marks IPv6 header:

![set_undef_after_mark_2](https://cloud.githubusercontent.com/assets/3772478/16573244/8394eb34-42ac-11e6-83a9-6c702898dfe7.png)

but `pkt` no longer contains IPv6 header, so that we have to change its type to `GNRC_NETTYPE_UNDEF`.

![set_undef_after_mark_3](https://cloud.githubusercontent.com/assets/3772478/16573255/9d123724-42ac-11e6-9705-f6065f7c6d4e.png)